aplib.net-demo/Assets/Scripts/Player/ShowPath.cs

```diff
@@ -0,0 +1,34 @@
+// ShowGoldenPath
+
+using UnityEngine;
+using UnityEngine.AI;
+
+public class ShowGoldenPath : MonoBehaviour
+{
+    public Transform target;
+    private float elapsed;
+    private NavMeshPath path;
+
+    private void Start()
+    {
+        path = new NavMeshPath();
+        elapsed = 0.0f;
+    }
+
+    private void Update()
+    {
+        // Update the way to the goal every second.
+        elapsed += Time.deltaTime;
+        if (elapsed > 1.0f)
+        {
+            elapsed -= 1.0f;
```

Comment:
I'm not sure, but if you want to do this every second don't you want to set elapsed to 0 instead of subtracting -1.0f? Like, if the system were to lag will it count further without executing the code or is that not a concern because it will always do this?

Reply:
Great point! This was just some debugging code copied from the unity site, so I did not really look at it as it worked, but great find!